### PR TITLE
flake.lock: Update

### PR DESCRIPTION
**Pull request by GaetanLepage**
`NuschtSearch` fails to build against the latest nixpkgs.

### CODE REVIEW

**Comment by MarcelCoding** [comment]
Thanks for the catch, it seems like a pnpm update resulted in a different hash.
Sadly the hash is therefore now broken with 24.05... @SuperSandro2000 Maybe pnpm should be backported upstream?

File: nix/frontend.nix [hunk 1]
```diff
@@ -21,7 +21,7 @@ stdenv.mkDerivation (finalAttrs: {
 
   pnpmDeps = pnpm.fetchDeps {
     inherit (finalAttrs) pname version src;
-    hash = "sha256-gTeGcV7zwZhIQIZ99Szet1tdKWVdEyPlXmilgZfJ/ko=";
+    hash = "sha256-4ri74PmqwWvuHTgrk3nnP/+RgBfl8GzA1mwBR1VJjTk=";
```
Comment (MarcelCoding) [inline nix/frontend.nix:24]:
```suggestion
    hash =
      if lib.versionAtLeast pnpm.version "9.12.1"
      then "sha256-4ri74PmqwWvuHTgrk3nnP/+RgBfl8GzA1mwBR1VJjTk="
      else "sha256-gTeGcV7zwZhIQIZ99Szet1tdKWVdEyPlXmilgZfJ/ko=";
```

**Comment by GaetanLepage** [comment]
```
>  ERR_PNPM_NO_OFFLINE_TARBALL  A package is missing from the store but cannot download it in offline mode. The missing package may be downloaded from https://registry.npmjs.org/js-yaml/-/js-yaml-4.1.0.tgz.
```

**Comment by MarcelCoding** [comment]
Yeah, already saw that - it seems like pnpm is not the reason for the different hash. (Our [demo](https://github.com/NuschtOS/search.nuschtos.de) instance uses an older version of nixpkgs which generates the old hash bus uses pnpm 9.12.1 too)

File: nix/frontend.nix [hunk 2]
```diff
@@ -21,7 +21,10 @@ stdenv.mkDerivation (finalAttrs: {
 
   pnpmDeps = pnpm.fetchDeps {
     inherit (finalAttrs) pname version src;
-    hash = "sha256-gTeGcV7zwZhIQIZ99Szet1tdKWVdEyPlXmilgZfJ/ko=";
+    hash =
+      if lib.versionAtLeast pnpm.version "9.12.1"
```
Comment (MarcelCoding) [inline nix/frontend.nix:25]:
```suggestion
      if lib.versionAtLeast pnpm.version "9.12.2"
```
Sorry I've messed that up. I've only looked at the search.nixos.org, and it probably still was on the outdated commit. The hash broke with 9.12.2 and not 9.12.1.

**Comment by MarcelCoding** [comment]
The cloudflare pages error is only because you created a fork and therefore don't get access the secrets in the CI.

File: nix/frontend.nix [hunk 3]
```diff
@@ -21,7 +21,10 @@ stdenv.mkDerivation (finalAttrs: {
 
   pnpmDeps = pnpm.fetchDeps {
     inherit (finalAttrs) pname version src;
-    hash = "sha256-gTeGcV7zwZhIQIZ99Szet1tdKWVdEyPlXmilgZfJ/ko=";
+    hash =
+      if lib.versionAtLeast pnpm.version "9.12.2"
+      then "sha256-4ri74PmqwWvuHTgrk3nnP/+RgBfl8GzA1mwBR1VJjTk="
```
Comment (SuperSandro2000) [inline nix/frontend.nix:26]: Not sure what the update script is now doing. 😅

Comment (MarcelCoding) [reply nix/frontend.nix:26]: hahah ;D its having fun

Comment (SuperSandro2000) [reply nix/frontend.nix:26]: I think we are just doing to overlay pnpm at this point 😅

Comment (SuperSandro2000) [reply nix/frontend.nix:26]: https://github.com/NuschtOS/search/pull/142

**Pull request merged**